code/parse/sexp.cpp

```diff
@@ -30111,8 +30082,9 @@ char *CTEXT(int n)
 	char *current_argument; 
 
 	Assertion(n >= 0 && n < Num_sexp_nodes, "Passed an out-of-range node index (%d) to CTEXT!", n);
-	if ( n < 0 ) {
-		return NULL;
+	if ( n < 0 || n >= Num_sexp_nodes ) {
+		// we should make CTEXT return const char*, but that's for another PR
+		return const_cast<char*>("!INVALID!");
```

Comment:
The returned string should be more specific (e.g. `INVALID CTEXT NODE INDEX SPECIFIED`) so that we immediately see where this comes from if we ever have issues with it.

Reply:
Ok. Done.